### PR TITLE
update alpine ascent and friends

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 1]
```diff
@@ -26,34 +34,36 @@ class Ascent(Package):
     simulations."""
 
     homepage = "https://github.com/Alpine-DAV/ascent"
-    git      = "https://github.com/Alpine-DAV/ascent.git"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]: Undo this change.

File: var/spack/repos/builtin/packages/ascent/package.py [hunk 2]
```diff
@@ -26,34 +34,36 @@ class Ascent(Package):
...
 
     maintainers = ['cyrush']
 
-    version('develop', branch='develop', submodules=True)
+    version('develop',
+            git='https://github.com/Alpine-DAV/ascent.git',
+            branch='develop',
+            submodules=True,
+            preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ascent/package.py]: preferred isn't necessary if there is only one version.

File: var/spack/repos/builtin/packages/conduit/package.py [hunk 3]
```diff
@@ -30,7 +37,7 @@ class Conduit(Package):
     url      = "https://github.com/LLNL/conduit/releases/download/v0.3.0/conduit-v0.3.0-src-with-blt.tar.gz"
     git      = "https://github.com/LLNL/conduit.git"
 
-    version('master', branch='master', submodules=True)
+    version('master', branch='master', submodules=True, preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/conduit/package.py]: I don't think an unstable branch should ever be the default over a stable release.

File: var/spack/repos/builtin/packages/mfem/package.py [hunk 4]
```diff
@@ -178,7 +178,7 @@ class Mfem(Package):
     depends_on('unwind', when='+libunwind')
     depends_on('zlib', when='+gzstream')
     depends_on('gnutls', when='+gnutls')
-    depends_on('conduit@0.3.1:', when='+conduit')
+    depends_on('conduit@master:', when='+conduit')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mfem/package.py]: `master` is the oldest version in the package. With this, you're saying that the version doesn't matter. I think you want: `@0.3.1:,master`

File: var/spack/repos/builtin/packages/vtkh/package.py [hunk 5]
```diff
@@ -15,31 +35,56 @@ class Vtkh(Package):
 
     homepage = "https://github.com/Alpine-DAV/vtk-h"
     git      = "https://github.com/Alpine-DAV/vtk-h.git"
-
-    version('master', branch='master', submodules=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vtkh/package.py]: Why delete this version?

File: var/spack/repos/builtin/packages/vtkh/package.py [hunk 6]
```diff
@@ -15,31 +35,56 @@ class Vtkh(Package):
...
+    variant("shared", default=True, description="Build vtk-h as shared libs")
     variant("mpi", default=True, description="build mpi support")
-    variant("tbb", default=True, description="build tbb support")
+    variant("tbb", default=False, description="build tbb support")
     variant("cuda", default=False, description="build cuda support")
+    variant("openmp", default=(sys.platform != 'darwin'),
+            description="build openmp support")
 
-    depends_on("cmake")
+    depends_on("cmake@3.8.2:")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/vtkh/package.py]: `type='build'`?

File: var/spack/repos/builtin/packages/conduit/package.py (hunk 3 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/conduit/package.py]: I am planning on an new release for conduit, 0.3.1 is almost two years old. 
That said, these packages require master -- and most consumers are using master.

File: var/spack/repos/builtin/packages/mfem/package.py (hunk 4 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/mfem/package.py]: that makes sense, thanks for clarifying. I think I just want to allow master to be built. 
The problem is master is actually newer and @0.3.1: did not allow it to build.

Sounds like `@0.3.1:,master` will do it?

File: var/spack/repos/builtin/packages/ascent/package.py (hunk 1 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/ascent/package.py]: just to clarify, I need to add this back in and remove git from the version command?

File: var/spack/repos/builtin/packages/ascent/package.py (hunk 2 above)
Comment (cyrush) [reply var/spack/repos/builtin/packages/ascent/package.py]: thanks, I removed another now broken version thats why its still there.  
Does it hurt to keep it?

File: var/spack/repos/builtin/packages/ascent/package.py (hunk 1 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/ascent/package.py]: Yep. You don't _have_ to, but it's easier to have a single package-level git.

File: var/spack/repos/builtin/packages/mfem/package.py (hunk 4 above)
Comment (adamjstewart) [reply var/spack/repos/builtin/packages/mfem/package.py]: Yep

**Comment by cyrush** [comment]
@adamjstewart thanks for the feedback, I think i address the major issues, let me know if you have any more concerns

File: var/spack/repos/builtin/packages/vtkm/package.py [hunk 7]
```diff
@@ -51,7 +52,11 @@ def cmake_args(self):
         with working_dir('spack-build', create=True):
             options = ["../",
                        "-DVTKm_ENABLE_TESTING:BOOL=OFF"]
-
+            # shared vs static libs
+            if "+shared" in spec:
+                options.append('-DBUILD_SHARED_LIBS=ON')
+            else:
+                options.append('-DBUILD_SHARED_LIBS=OFF')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/vtkm/package.py]: I dunno if you find it clearer, but FYI, this works (b/c CMake is case-insensitive and understands `True` and `False`):

```python
options.append('-DBUILD_SHARED_LIBS=%s' % ('+shared' in spec))
```

**Comment by tgamblin** [comment]
@cyrush: This looks fine to me, but do you want to add a `BltPackage` class that subclasses CMakePackage, so that we can consolidate the `create_host_config` logic and any of the other boilerplate required for mapping Spack stuff to BLT?

**Comment by cyrush** [comment]
@tgamblin hmm, well host config isn't unqiue to BLT -- it's strategy that could be used for any cmake package.  Yeah it's a good amount of code.  Sounds like a good thing to ponder. I am a fan of still using the generic base class package.  Less mystery about order of operations, more control -- even if it's a bit more verbose, its self contained.

**Comment by tgamblin** [comment]
Ok.  If it makes sense and is general, we could also add that stuff to CmakePackage.

**Comment by cyrush** [comment]
@tgamblin thanks!
